### ISSUE TITLE
GCC build fails with Xcode 6.3 on Yosemite

### ISSUE THREAD

**Issue by dcarley** [alphagov/gds-boxen#156]
I upgraded to Yosemite at last. However it seems that I picked the wrong day to do it, because Xcode 6.3 was also released and GCC no longer builds against it. Boxen runs `brew boxen-install homebrew/versions/gcc48` which fails with the following error:
```
Comparing stages 2 and 3
warning: gcc/cc1-checksum.o differs
warning: gcc/cc1obj-checksum.o differs
warning: gcc/cc1objplus-checksum.o differs
warning: gcc/cc1plus-checksum.o differs
Bootstrap comparison failure!
gcc/asan.o differs
gcc/c/c-decl.o differs
gcc/cfg.o differs
gcc/coverage.o differs
gcc/cp/class.o differs
gcc/cp/semantics.o differs
gcc/cp/tree.o differs
gcc/dse.o differs
gcc/objc/objc-act.o differs
gcc/objcp/objcp-act.o differs
gcc/tree-ssa-ccp.o differs
gcc/tree-ssa-coalesce.o differs
gcc/tree-ssa-pre.o differs
gcc/tree-ssa-threadupdate.o differs
gcc/valtrack.o differs
make[2]: *** [compare] Error 1
make[1]: *** [stage3-bubble] Error 2
make: *** [bootstrap] Error 2
```

Dowgrading to Xcode 6.1.1 worked. I didn't try 6.2 because I was pretty sure that other people were successfully using 6.1.

I don't suppose that this is a problem with boxen/puppet-gcc and I'm not sure where else I'd record such a bug, so for the moment I'm putting it here to make other users of gds-boxen aware.

**Comment by samjsharpe** [comment]
Just out of interest, what version of gcc is it trying to install? I have 4.8.3-boxen2 installed and I presume that was compiled with Xcode 6.2. I can try unlinking gcc and reinstalling, but without fudging that tries to install 4.8.4 (not a boxen version)

**Comment by dcarley** [comment]
That version - `4.8.3-boxen2`.

What about if you use the same command as Boxen which is `brew boxen-install homebrew/versions/gcc48`?

**Comment by samjsharpe** [comment]
`brew boxen-install homebrew/versions/gcc48` is what was installing 4.8.4 for me. I'll have a look when I get a sec and see if my Formula is different - my homebrew tree looks clean and appears to be at Homebrew/homebrew@13a086f - by fudging I was referring to resetting my homebrew tree to try to force that version to install as I don't know how else you force an install of a specific version.

**Comment by TessMyers** [comment]
Trashing my recently downloaded Xcode 6.3.1 and replacing it with [Xcode 6.2](https://developer.apple.com/downloads/index.action) fixed this problem for me, so Xcode is fine up to 6.2.

**Comment by dcarley** [comment]
Thanks, Tess!

**Comment by bradleywright** [comment]
I merged #162 so this should be fixed.
